packages/atomic/src/components/search/search-box-suggestions/suggestions-common.ts

```diff
@@ -3,41 +3,136 @@ import {VNode} from '@stencil/core';
 import {buildCustomEvent} from '../../../utils/event-utils';
 import {closest} from '../../../utils/utils';
 import {Bindings} from '../atomic-search-interface/atomic-search-interface';
+
+/**
+ * Element which will be rendered in the list of suggestions.
+ */
 export interface SearchBoxSuggestionElement {
+  /**
+   * Stable identity which enables Stencil to reuse DOM elements for better performance.
+   * The best way to pick a key is to use a string that uniquely identifies that list item among its siblings (often your data will already have IDs).
+   */
   key: string;
+  /**
+   * Rendered content of the element.
+   */
   content: Element | VNode;
+  /**
+   * Hook called when the selection is selected.
+   * @param e DOM event.
+   */
   onSelect?(e: Event): void;
-
+  /**
+   * The query associated with the suggestion, which will become the suggested query that would be sent if the search is executed.
+   */
   query?: string;
+  /**
+   * For improved accessibility, provide this property with additional information.
+   * https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label
+   */
   ariaLabel?: string;
+  /**
+   * Adds a specific shadow part attribute that can be selected with the CSS ::part pseudo-element.
+   * https://developer.mozilla.org/en-US/docs/Web/CSS/::part
+   */
   part?: string;
+  /**
+   * Hide the suggestion if it's the last in the list.
+   */
   hideIfLast?: boolean;
 }
+
+/**
+ * List of suggestions that will be displayed along other lists (e.g recent queries) when the search box's input is selected.
+ */
 export interface SearchBoxSuggestions {
+  /**
+   * The position of the list of the suggestions relative to the others.
+   */
   position: number;
+  /**
+   * Whether the suggestions should be listed in the right or left panel.
+   */
   panel?: 'left' | 'right';
+  /**
+   * Method that returns the list of elements which will be rendered in the list of suggestions.
+   */
   renderItems(): SearchBoxSuggestionElement[];
+  /**
+   * Hook called when the user changes the search box's input value.
+   */
   onInput?(): Promise<unknown> | void;
+  /**
+   * Hook called when the suggested query changes.
+   * @param q The new suggested query.
+   */
   onSuggestedQueryChange?(q: string): Promise<unknown> | void;
 }
 
+/**
+ * Event sent from the registered query suggestions to the parent search box.
+ */
 export type SearchBoxSuggestionsEvent = (
+  /**
+   * The bindings passed from the search box to the suggestions.
+   */
   bindings: SearchBoxSuggestionsBindings
 ) => SearchBoxSuggestions;
 
+const searchBoxElements = ['atomic-search-box', 'atomic-insight-search-box'];
+
+/**
+ * The bindings passed from the search box to the suggestions.
+ */
 export interface SearchBoxSuggestionsBindings extends Bindings {
+  /**
+   * The unique id of the search box.
+   */
   id: string;
+  /**
+   * Whether the search box is standalone.
```

Comment:
I think we should link to some doc about standalone search boxes.

Reply:
```suggestion
   * Whether the search box is standalone.
   * https://docs.coveo.com/en/atomic/latest/usage/ssb/
```
Referencing an Atomic article seems to make the most sense to me. Even though this page does not directly address what a standalone search box is, it should. I will add a paragraph to the start of this page.